File: Part 2 - Regression/Section 7 - Support Vector Regression (SVR)/svr.R
# Support Vector regression

#Importing dataset
dataset = read.csv('Position_Salaries.csv')
dataset = dataset[2:3]

#Fitting the dataset to SVR
#install.packages('e1071')
library(e1071)
regressor = svm(formula = Salary ~ .,
                data = dataset,
                type = 'eps-regression')

#Prediciting a new resultset
y_pred = predict(regressor, data.frame(Level = 6.5))

#Visualizing the SVR results
library(ggplot2)
ggplot() +
  geom_point(aes(x = dataset$Level, y = dataset$Salary),
             colour = 'red')+
  geom_line(aes(x = dataset$Level, y = predict(regressor, newdata = dataset)),
            colour = 'blue')+
  ggtitle('Truth vs Bluff (SVR)')+
  xlab('Level')+
  ylab('Salary')

File: Part 2 - Regression/Section 7 - Support Vector Regression (SVR)/svr.py
#SVR(Support Vector regression)

#Importing the libraries
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

#Importing the dataset
dataset = pd.read_csv('Position_Salaries.csv')
X = dataset.iloc[:, 1:2].values
y = dataset.iloc[:, 2].values

#Feature Scaling
from sklearn.preprocessing import StandardScaler
sc_X = StandardScaler()
sc_y = StandardScaler()
X = sc_X.fit_transform(X)
y = y.reshape(-1,1)
y = sc_y.fit_transform(y)

#Fitting SVR to dataset
from sklearn.svm import SVR
regressor = SVR(kernel = 'rbf')
regressor.fit(X, y)

#Predicting the new resultset
y_pred = sc_y.inverse_transform(regressor.predict(sc_X.transform(np.array([[6.5]]))))

#Visualizing the SVR Result
X_grid = np.arange(min(X), max(y), 0.1)
X_grid = X_grid.reshape(len(X_grid), 1)
plt.scatter(X, y, color='red')
plt.plot(X_grid, regressor.predict(X_grid), color='blue')
plt.title('Truth vs Bluff (SVR)')
plt.xlabel('Position Level')
plt.ylabel('Salary')
plt.show()